src/openforms/tests/e2e/test_input_validation.py

```diff
@@ -292,3 +317,209 @@ def test_max_date_fixed_value(self):
             api_value="2025-01-01",
             expected_ui_error="De opgegeven datum ligt te ver in de toekomst.",
         )
+
+
+class SingleCheckboxTests(ValidationsTestCase):
+
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        await expect(
+            page.get_by_role("checkbox", name="Required checkbox")
+        ).not_to_be_checked()
+
+    def test_required_field(self):
+        component: Component = {
+            "type": "checkbox",
+            "key": "requiredCheckbox",
+            "label": "Required checkbox",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            api_value=False,
+            expected_ui_error="Het verplichte veld Required checkbox is niet ingevuld.",
+        )
+
+
+class SingleCurrencyTests(ValidationsTestCase):
+    def test_required_field(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required currency is niet ingevuld.",
+        )
+
+    def test_min_value(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"min": 10.7},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=2,
+            expected_ui_error="De waarde moet 10.7 of groter zijn.",
+        )
+
+    def test_max_value(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"max": 15},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=50,
+            expected_ui_error="De waarde moet 15 of kleiner zijn.",
+        )
+
+
+class SingleMapTests(ValidationsTestCase):
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        page.wait_for_selector(
```

Comment:
shouldn't this be awaited?

```suggestion
        await page.wait_for_selector(
```